File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
            
        int leftIdx = 0;
        int rightIdx = numbers.length - 1;
        
        while(leftIdx < rightIdx)
        {
            int sum = numbers[leftIdx] + numbers[rightIdx];
            if(sum == target) return new int[]{leftIdx+1, rightIdx+1};
            else if(sum < target) leftIdx++;
            else if(sum > target) rightIdx--;
        }
        
        return null;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left,right = 0, len(numbers) - 1
        while(left < right):
            le,ri = numbers[left], numbers[right]
            if(target == le + ri):
                return [left+1, right+1]
            elif(target > le+ri):
                left += 1
            elif(target < le+ri):
                right -= 1

        return [0,0]